src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -365,9 +465,9 @@ public ImageInfo Identify(BufferedReadStream stream, CancellationToken cancellat
             }
 
             EOF:
-            if (this.header.Width == 0 && this.header.Height == 0)
+            if (this.header is { Width: 0, Height: 0 })
```

Comment:
I can't stand this syntax. Pattern matching should be used to simplify readability, not obfuscate it.